diff --git a/Makefile.flags b/Makefile.flags
index 6f6142c..f93afde 100644
--- a/Makefile.flags
+++ b/Makefile.flags
@@ -51,7 +51,7 @@ CFLAGS += $(call cc-option,-fno-builtin-strlen -finline-limit=0 -fomit-frame-poi
 # -fno-guess-branch-probability: prohibit pseudo-random guessing
 # of branch probabilities (hopefully makes bloatcheck more stable):
 CFLAGS += $(call cc-option,-fno-guess-branch-probability,)
-CFLAGS += $(call cc-option,-funsigned-char -static-libgcc,)
+CFLAGS += $(call cc-option,-funsigned-char,)
 CFLAGS += $(call cc-option,-falign-functions=1 -falign-jumps=1 -falign-labels=1 -falign-loops=1,)
 # Defeat .eh_frame bloat (gcc 4.6.3 x86-32 defconfig: 20% smaller busybox binary):
 CFLAGS += $(call cc-option,-fno-unwind-tables,)
diff --git a/include/platform.h b/include/platform.h
index 50365a3..c7c69fc 100644
--- a/include/platform.h
+++ b/include/platform.h
@@ -157,9 +157,9 @@
    || defined(__APPLE__)
 # include <sys/resource.h>  /* rlimit */
 # include <machine/endian.h>
-# define bswap_64 __bswap64
-# define bswap_32 __bswap32
-# define bswap_16 __bswap16
+# define bswap_64 __builtin_bswap64
+# define bswap_32 __builtin_bswap32
+# define bswap_16 __builtin_bswap16
 #else
 # include <byteswap.h>
 # include <endian.h>
@@ -480,6 +480,8 @@ typedef unsigned smalluint;
 #endif
 
 #if defined(__APPLE__)
+# undef HAVE_MEMRCHR
+# undef HAVE_MEMPCPY
 # undef HAVE_STRCHRNUL
 #endif
 
diff --git a/scripts/trylink b/scripts/trylink
index bb6b2de..fec6d73 100755
--- a/scripts/trylink
+++ b/scripts/trylink
@@ -89,10 +89,10 @@ if ! check_cc "-Wl,--sort-section,alignment"; then
     SORT_SECTION=""
 fi
 
-START_GROUP="-Wl,--start-group"
-END_GROUP="-Wl,--end-group"
+START_GROUP=""
+END_GROUP=""
 INFO_OPTS() {
-	echo "-Wl,--warn-common -Wl,-Map,$EXE.map -Wl,--verbose"
+	echo ""
 }
 
 # gold may not support --sort-common (yet)
diff --git a/shell/shell_common.c b/shell/shell_common.c
index e0582ad..e088c1e 100644
--- a/shell/shell_common.c
+++ b/shell/shell_common.c
@@ -328,7 +328,9 @@ struct limits {
 static const struct limits limits_tbl[] = {
 	{ RLIMIT_CORE,		9,	"core file size (blocks)" }, // -c
 	{ RLIMIT_DATA,		10,	"data seg size (kb)" },      // -d
+#ifdef RLIMIT_NICE
 	{ RLIMIT_NICE,		0,	"scheduling priority" },     // -e
+#endif
 	{ RLIMIT_FSIZE,		9,	"file size (blocks)" },      // -f
 #define LIMIT_F_IDX     3
 #ifdef RLIMIT_SIGPENDING
